be/src/storage/tablet_updates.cpp

```diff
@@ -1526,11 +1526,78 @@ Status TabletUpdates::compaction(MemTracker* mem_tracker) {
 
     Status st = _do_compaction(&info, true);
     if (!st.ok()) {
+        _last_compaction_failure_millis = UnixMillis();
         _compaction_running = false;
     }
+    _last_compaction_success_millis = UnixMillis();
```

Comment:
else {
    _last_compaction_success_millis = UnixMillis();
}

Reply:
> else { _last_compaction_success_millis = UnixMillis(); }

done, fix in https://github.com/StarRocks/starrocks/pull/3123/commits/b4f15d173b87b827760de418dc60fe7c3b1dc1ec